sdk/storage/Azure.Storage.Common/src/Shared/PartitionedUploader.cs

```diff
@@ -376,5 +440,169 @@ public struct Behaviors
                 partition.Dispose();
             }
         }
+
+        /// <summary>
+        /// Some streams will throw if you try to access their length so we wrap
+        /// the check in a TryGet helper.
+        /// </summary>
+        private static long? GetLengthOrDefault(Stream content)
+        {
+            if (content == null)
+            {
+                /* Returning 0 instead of default puts us on the quick and clean one-shot upload,
+                 * which produces more consistent fail state with how a 1-1 method on the convenience
+                 * layer would fail.
+                 */
+                return 0;
+            }
+            try
+            {
+                if (content.CanSeek)
+                {
+                    return content.Length;
+                }
+            }
+            catch (NotSupportedException)
+            {
+            }
+            return default;
+        }
+
+        #region Stream Splitters
+        /// <summary>
+        /// Partition a stream into a series of blocks buffered as needed by an array pool.
+        /// </summary>
+        private static async IAsyncEnumerable<StreamSlice> GetPartitionsAsync(
+            Stream stream,
+            long? streamLength,
+            long blockSize,
+            GetNextStreamPartition getNextPartition,
+            bool async,
+            [EnumeratorCancellation] CancellationToken cancellationToken)
+        {
+            // The minimum amount of data we'll accept from a stream before
+            // splitting another block.
+            long acceptableBlockSize = blockSize / 2;
```

Comment:
It seems there is no point in handwritten code where we validate user-specified block size > 0. That would likely cause failure at some other layer, and would be an existing bug in today's GA code (this logic previously existed and was moved in refactors). We can probably take the initial fetch of block size and assert it's > 0, and then clamp `acceptableBlockSize` to also be > 0. While it may change how we fail in such a scenario, we'll still fail in some regard. Additionally, it's extremely unlikely customers are currently passing in values <= 1 in production code, so such a change should not pose a risk to anyone.

Reply:
I'd add such validation then and let it fail fast and clean.

Reply:
Was incorrect about how users can set block size. They can set whatever they want, but if they set a non-positive number then we ignore it and use our defaults. So at this point, `blockSize` will always be positive and there is no need to insert a fail-fast check. Will add `Min(1, blockSize / 2)` to this line in case someone passes block size of 1, but that's more of a sanity check.